posthog/temporal/batch_exports/pre_export_stage.py

```diff
@@ -526,7 +530,14 @@ async def _write_batch_export_record_batches_to_s3(
             query_id = uuid.uuid4()
             await logger.ainfo(f"Executing query with ID = {query_id}")
             try:
-                await client.execute_query(query, query_parameters=query_parameters, query_id=str(query_id))
+                await client.execute_query(
+                    query, query_parameters=query_parameters, query_id=str(query_id), timeout=300
+                )
+            except ClickHouseClientTimeoutError:
+                status = ClickHouseQueryStatus.RUNNING
+                while status == ClickHouseQueryStatus.RUNNING:
+                    status = await client.acheck_query(str(query_id), raise_on_error=True)
```

Comment:
might be good to have some (possibly customisable) delay before re-checking?

Reply:
yeah good point